### ISSUE TITLE
where's the actual migration explantion?

### ISSUE THREAD

**Issue by Pomax** [tgriesser/knex#992]
The [main docs](http://knexjs.org/#Installation-migrations) say "Read the Migrations section for more information and a full list of configuration options", followed by an example file. The [migrations section](http://knexjs.org/#Migrations) then doesn't even explain all the things listed in the example.

What does the `tablename` do? How do I load migrations from file, in sequence? There's almost no information in the migration section.

**Comment by rhys-vdw** [comment]
Related to #563.

A PR resolving this issue would be appreciated.

**Comment by harriha** [comment]
Tried to improve this a while back, in #956

**Comment by Pomax** [comment]
super nice! If there was also an example migration and seeding, that PR would pretty much be perfect.

**Comment by rhys-vdw** [comment]
@harriha I had not seen this. Will review soon.

**Comment by harriha** [comment]
@Pomax Thanks - what comes to examples in general, it's often useful to take a look at tests of a project. For example, migration tests can be found from https://github.com/tgriesser/knex/tree/master/test/integration/migrate, might help at least a bit. But true, I guess the docs themselves could always be better.

@rhys-vdw Thanks, please don't hesitate to throw the PR back to me if there's issues with it.

**Comment by Pomax** [comment]
@harriha, yeah having documentation in addition to the code (the test/integration code has no comments, for instance) is super useful.

**Comment by mercmobily** [comment]
Thanks for bringing this up back in 2015! You're absolutely right that the migration documentation was lacking at that time, particularly around configuration options like `tableName` and how to properly configure migrations.

I'm happy to report that we've made significant improvements to the migration documentation since then. Looking at the current docs, we now have comprehensive coverage of:

- The `tableName` configuration option (now called `tableName` in the Migration API section) which specifies the table name used for storing migration state (defaults to `knex_migrations`)
- All the migration configuration options including `directory`, `extension`, `schemaName`, `disableTransactions`, and many others
- Detailed explanations of the Migration CLI commands
- Complete knexfile.js configuration examples
- How migration sequencing works and how to load migrations from files

The documentation improvements you were asking for were actually implemented through PR #956 by @harriha, which merged in March 2016 and significantly enhanced the migration docs with descriptions for each config option.

That said, documentation is always evolving and we're constantly working to make it better. There are still some related open issues like #563 (Document knexfile options), #1318 (extension family documentation), and #1656 (async nature of APIs) that we continue to address.

If you have specific areas of the current migration documentation that you feel could be clearer or more comprehensive, I'd encourage you to open a new issue with those specific suggestions. We really appreciate feedback that helps us improve the developer experience!

**Related Issues:**
- #563 - Document knexfile options 
- #956 - Improve tests and docs for migrate API (merged)
- #1318 - No documentation for the extension family of migrations
- #1656 - Documentation: async, promise returning nature of migration and seed APIs
